tests/func/experiments/test_remote.py

```diff
@@ -151,7 +151,7 @@ def test_pull(tmp_dir, scm, dvc, git_downstream, exp_stage, use_url):
 
     git_downstream.scm.remove_ref(str(ref_info))
 
-    downstream_exp.pull(remote, str(ref_info))
+    downstream_exp.pull(remote, str(ref_info), pull_cache=True)
```

Comment:
Is there a reason for this change? We have a dedicated end-to-end test for pushing and pulling cache (`test_push_pull_cache`)

Reply:
for the #6421 ,

Reply:
So that bug only occurred when we pull cache, but it's not obvious why setting `pull_cache` here should affect whether or not using a full refname works or not.

What we really want for #6421 is a unit test that verifies that the get_exp_ref utils function always returns an `ExpRefInfo`, whether it is passed a shortened name or a full `refs/...` string.

Reply:
Yes but there are two problems.
1. the original function had been removed.
2. we already have a type examine for the new function .
So a functional test or no test and let the issue submitter test him/herself.

Reply:
Type checking isn't a replacement for a unit test, the new function (`resolve_exp_ref`) should get a unit test instead of this `pull_cache` change.